src/main/scala/org/ergoplatform/nodeView/history/storage/modifierprocessors/HeadersProcessor.scala

```diff
@@ -257,34 +258,77 @@ trait HeadersProcessor extends ToDownloadProcessor with ScorexLogging with Score
 
   protected def heightIdsKey(height: Int): ByteArrayWrapper = ByteArrayWrapper(Algos.hash(Ints.toByteArray(height)))
 
+
+  private val EIP37VotingParameter: Byte = 8 // todo: should be 6 after testing , and 6 = 2100 in config
+
+  private val eip37Key = Blake2b256.hash("eip37 activation height")
+
+  private def storeEip37ActivationHeight(eip37ActivationHeight: Int) = {
+    historyStorage.insert(eip37Key, Ints.toByteArray(eip37ActivationHeight))
+  }
+
+  private def eip37VotedOn: Option[Int] = {
+    historyStorage.get(scorex.util.bytesToId(eip37Key)).map(Ints.fromByteArray)
+  }
+
   /**
     * Calculate difficulty for the next block
     *
     * @param parent - latest block
     * @return - difficulty for the next block
     */
   def requiredDifficultyAfter(parent: Header): Difficulty = {
-    if (parent.height == settings.chainSettings.voting.version2ActivationHeight || parent.height + 1 == settings.chainSettings.voting.version2ActivationHeight) {
-      // Set difficulty for version 2 activation height (where specific difficulty is needed due to PoW change)
-      settings.chainSettings.initialDifficultyVersion2
+    val parentHeight = parent.height
+
+    // todo: this EIP-37 activation checking code could be removed after activation
+    if (settings.chainSettings.isMainnet &&
+        parentHeight > 843776 &&
```

Comment:
`843776` and `128` are good candidates for named constants.

Reply:
Also 8192 and 512. The name can also document the meaning.

Reply:
done